offer-manager-domain/src/main/groovy/life/qbic/business/offers/content/CreateOfferContent.groovy

```diff
@@ -0,0 +1,208 @@
+        groupProductItems(offer.items)
+        List<OfferItem> dataManagementOfferItems = dataManagementItems.collect{createProductItemToOfferItem(it)}
+        List<OfferItem> dataAnalysisOfferItems = dataAnalysisItems.collect{createProductItemToOfferItem(it)}
+        List<OfferItem> dataGenerationOfferItems = dataGenerationItems.collect{createProductItemToOfferItem(it)}
+
+        offerContent.dataGenerationItems(dataGenerationOfferItems)
+        .dataAnalysisItems(dataAnalysisOfferItems)
+        .dataManagementItems(dataManagementOfferItems)
+
+        double overheadsDA = calculateOverheadSum(dataAnalysisOfferItems)
+        double overheadsDG = calculateOverheadSum(dataGenerationOfferItems)
+        double overheadsPMandDS = calculateOverheadSum(dataManagementOfferItems)
+
+        offerContent.overheadsDataAnalysis(overheadsDA)
+        .overheadsDataGeneration(overheadsDG)
+        .overheadsProjectManagementAndDataStorage(overheadsPMandDS)
+        .overheadTotal(offer.overheadSum)
+
+        offerContent.netDataAnalysis(calculateNetSum(dataAnalysisOfferItems))
+        .netDataGeneration(calculateNetSum(dataGenerationOfferItems))
+        .netProjectManagementAndDataStorage(calculateNetSum(dataManagementOfferItems))
+        .netCost(offer.totalNetPrice)
+
+        offerContent.totalVat(offer.taxCosts)
+        .vatRatio(offer.determineTaxCost())
+        .totalCost(offer.totalCosts)
+
+        OfferContent offerContentFinal = offerContent.build()
+
+        output.createdOfferContent(offerContentFinal)
+    }
+
+    /**
+     * Helper method that calculates the NET price for a list of product items
+     * @param offerItems The product item list for which the NET is calculated
+     *  @return The net value for the given list of items
+     */
+    private static double calculateNetSum(List<OfferItem> offerItems) {
+        double netSum = 0
+        offerItems.each {
+            netSum += it.quantity * it.unitPrice
+        }
+        return netSum
+    }
+
+    /**
+     * Calculates the overhead sum of all product items
+     * @param offerItems Items for which the overheads are calculated
+     * @return The overhead price for all items
+     */
+    private double calculateOverheadSum(List<OfferItem> offerItems) {
+        double overheadSum = 0.0
+        offerItems.each {
+            overheadSum += it.quantity * it.unitPrice * overheadRatio
+        }
+        return overheadSum
+    }
+
+
+    private OfferItem createProductItemToOfferItem(ProductItem productItem){
+        Product product = productItem.product
+        double unitPrice = (affiliationCategory == AffiliationCategory.INTERNAL) ? product.internalUnitPrice : product.externalUnitPrice
+
+        OfferItem offerItem = new OfferItem.Builder(productItem.quantity, product.description, product.productName, unitPrice, productItem.quantityDiscount,
+                product.serviceProvider.name(), product.unit.name(), productItem.totalPrice).build()
```

Comment:
```suggestion
                product.serviceProvider.fullName, product.unit.name(), productItem.totalPrice).build()
```
If you want the offer to display "Quantitative Biology Center" instead of "QBIC" then you need to use the fullName property

Reply:
I know, for now its decided the the short name is sufficient